### PR TITLE
Add std/core to prelude if extern_prelude enabled

### LINKED ISSUE #50605: using extern-prelude does not allow use of std/core
Example ([play](https://play.rust-lang.org/?gist=dd59b53a3b5e7a03573314b919339e89&version=nightly&mode=debug)):

```rust
#![feature(extern_prelude)]

mod foo {
    pub fn test() {
        let x = std::cmp::min(2, 3);
        assert_eq!(x, 2);
    }
}

fn main() {
    foo::test();
}
```

I would expect this to work. (I presume it doesn't because cargo doesn't pass `--extern std`.)

cc @petrochenkov  @Manishearth

### PR DESCRIPTION
**Pull request by mbrubeck**
Fixes #50605

### CODE REVIEW

File: src/librustc_resolve/lib.rs [hunk 1]
```diff
@@ -1610,6 +1610,14 @@ impl<'a> Resolver<'a> {
         DefCollector::new(&mut definitions, Mark::root())
             .collect_root(crate_name, session.local_crate_disambiguator());
 
+        let mut extern_prelude: FxHashSet<Name> =
+            session.opts.externs.iter().map(|kv| Symbol::intern(kv.0)).collect();
+        if !attr::contains_name(&krate.attrs, "no_std") {
+            extern_prelude.insert(Symbol::intern("std"));
```
Comment (petrochenkov) [inline src/librustc_resolve/lib.rs]: This adds `std` to the prelude for `no_core` crates.
(The logic should mirror `fn maybe_inject_crates_ref` in `libsyntax/std_inject.rs`.)

Comment (petrochenkov) [reply src/librustc_resolve/lib.rs]: Additionally, should `core` be available in extern prelude by default (for not `no_std`/`no_core` crates)?
Also, should other stable crates from the standard distribution like `proc_macro` be available?
I'd say no for a start, they can always be added later.

Comment (mbrubeck) [reply src/librustc_resolve/lib.rs]: Fixed this to share the same logic as `maybe_inject_crates_ref`.

File: src/librustc_resolve/lib.rs [hunk 2]
```diff
@@ -1610,6 +1611,12 @@ impl<'a> Resolver<'a> {
         DefCollector::new(&mut definitions, Mark::root())
             .collect_root(crate_name, session.local_crate_disambiguator());
 
+        let mut extern_prelude: FxHashSet<Name> =
+            session.opts.externs.iter().map(|kv| Symbol::intern(kv.0)).collect();
+        for name in crates_to_inject(&krate) {
```
Comment (petrochenkov) [inline src/librustc_resolve/lib.rs]: I don't think we need to add `compiler_builtins` to the prelude though, like `crates_to_inject` does.

Comment (mbrubeck) [reply src/librustc_resolve/lib.rs]: Fixed.

**Comment by mbrubeck** [comment]
Rebased.

**Pull request merged**